Tests/CommandsTests/BuildCommandTests.swift

```diff
@@ -1330,6 +1330,77 @@ struct BuildCommandTestCases {
         }
     }
 
+    @Test(
+         .bug("https://github.com/swiftlang/swift-package-manager/issues/8844"),
+         arguments: SupportedBuildSystemOnPlatform,  BuildConfiguration.allCases
+     )
+     func swiftBuildQuietLogLevel(
+         buildSystem: BuildSystemProvider.Kind,
+         configuration: BuildConfiguration
+     ) async throws {
+         // GIVEN we have a simple test package
+         try await fixture(name: "Miscellaneous/SwiftBuild") { fixturePath in
+            //WHEN we build with the --quiet option
+            let (stdout, stderr) = try await executeSwiftBuild(
+                fixturePath,
+                configuration: configuration,
+                extraArgs: ["--quiet"],
+                buildSystem: buildSystem
+            )
+            // THEN we should not see any output in stderr
+             #expect(stderr.isEmpty)
+            // AND no content in stdout
+             #expect(stdout.isEmpty)
+        }
+    }
+
+    @Test(
+         .bug("https://github.com/swiftlang/swift-package-manager/issues/8844"),
+         arguments: SupportedBuildSystemOnPlatform,  BuildConfiguration.allCases
+     )
+     func swiftBuildQuietLogLevelWithError(
+         buildSystem: BuildSystemProvider.Kind,
+         configuration: BuildConfiguration
+     ) async throws {
+         // GIVEN we have a simple test package
+         try await fixture(name: "Miscellaneous/SwiftBuild") { fixturePath in
+             let mainFilePath = fixturePath.appending("main.swift")
+             try localFileSystem.removeFileTree(mainFilePath)
+             try localFileSystem.writeFileContents(
+                mainFilePath,
+                string: """
+                 print("done"
+                 """
+             )
+
+             //WHEN we build with the --quiet option
+             let error = await #expect(throws: SwiftPMError.self) {
+                 try await executeSwiftBuild(
+                    fixturePath,
+                    configuration: .debug,
+                    extraArgs: ["--quiet"],
+                    buildSystem: buildSystem
+                 )
+             }
+
+             if case .executionFailure(underlying: _, stdout: let stdout, stderr: let stderr) = error {
```

Comment:
issue (possibly-blocking): A more idiomatic way would be to do

```swift
            guard case SwiftPMError.executionFailure(_, let stdout, let stderr) = try #require(error) else {
                Issue.record("Incorrect error was raised.")
                return
            }
            if buildSystem == .swiftbuild {
                // expectation
            } else  {
                 // expectations
            }
````

NOTE: this comment applies to all similar instances